pkg/controlplane/reconcilers/lease_test.go

```diff
@@ -514,6 +524,7 @@ func TestLeaseRemoveEndpoints(t *testing.T) {
 			clientset := fake.NewSimpleClientset(test.initialState...)
 			epAdapter := NewEndpointsAdapter(clientset.CoreV1(), clientset.DiscoveryV1())
 			r := NewLeaseEndpointReconciler(epAdapter, fakeLeases)
+			r.StopReconciling()
 			err = r.RemoveEndpoints(test.serviceName, netutils.ParseIPSloppy(test.ip), test.endpointPorts)
```

Comment:
does this no affect the other tests? 
we may add more test cases to exercise the races you mentioned during the review

Reply:
`r.StopReconciling()` has no side effect to current tests because the precondition for executing `TestLeaseRemoveEndpoints` function is that the apiserver has already been shutdown. It's better to add test case for the scenario of apiserver start up.